src/main/java/capstone/mfslbackend/service/UserService.java

```diff
@@ -0,0 +1,51 @@
+package capstone.mfslbackend.service;
+
+import capstone.mfslbackend.model.Authority;
+import capstone.mfslbackend.model.SecurityUser;
+import capstone.mfslbackend.model.User;
+import capstone.mfslbackend.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class UserService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findUserByUsername(username);
+
+        if (user == null) {
+            throw new UsernameNotFoundException("No user found for username: " + username);
+        }
+
+        return new SecurityUser(user);
+    }
+
+    public User createUser(String username, String password) {
+        // todo check valid password, could do it in fe maybe
+
+        User user = new User();
+        user.setUsername(username);
+        user.setPassword(passwordEncoder.encode(password));
+        user.getAuthorities().add(Authority.LEAGUE_MEMBER);
```

Comment:
@seb8stien1 @linamoufarreh Right now every new user is made with the league member role. Based on how you guys implement the member and admin roles, change this accordingly.